File: solutions/typescript/VB_T3_012_pair_sum.ts
// VB-T3-012: Pair Sum -- TypeScript baseline
type Pair = { tag: "MkPair"; first: number; second: number };

function pairSum(a: number, b: number): number {
  const p: Pair = { tag: "MkPair", first: a, second: b };
  return p.first + p.second;
}

console.assert(pairSum(3, 4) === 7);
console.assert(pairSum(0, 0) === 0);
console.assert(pairSum(-5, 5) === 0);
console.assert(pairSum(10, -3) === 7);
console.assert(pairSum(100, 200) === 300);

File: solutions/python/VB_T3_012_pair_sum.py
"""VB-T3-012: Pair Sum -- Python baseline."""


class Pair:
    pass


class MkPair(Pair):
    __match_args__ = ("first", "second")

    def __init__(self, first: int, second: int) -> None:
        self.first = first
        self.second = second


def pair_sum(a: int, b: int) -> int:
    p = MkPair(a, b)
    match p:
        case MkPair(x, y):
            return x + y
    return 0


if __name__ == "__main__":
    assert pair_sum(3, 4) == 7
    assert pair_sum(0, 0) == 0
    assert pair_sum(-5, 5) == 0
    assert pair_sum(10, -3) == 7
    assert pair_sum(100, 200) == 300
    print("VB-T3-012 ok")
